miraclelinux/meta-debian-extended
# bind: Update version from 9.11.5.P4+dfsg-5.1+deb10u7 to 9.11.5.P4+dfsg-5.1+deb10u8

# Purpose of pull request

This version fixes following CVEs.
- CVE-2022-2795
- CVE-2022-38177
- CVE-2022-38178

DLA: https://www.debian.org/lts/security/2022/dla-3138


# Test
## How to test

1. build bind package
2. Run qemu image
3. Check bind server is working
4. Run some command that provided by bind-utils package

## Test result

Build result.

```
┌──build@931e89c96a90:~/emlinux/latest-dev/build
└─$ bitbake bind
Loading cache: 100% |##################################################################################################################################################################| Time: 0:00:00
Loaded 2360 entries from dependency cache.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "ubuntu-18.04"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.6"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "HEAD:c69abc6f79aefbad60ab1ba97cc1792b6af4b4dd"
meta-debian-extended = "update-bind-9.11.5.P4+dfsg-5.1+deb10u8:6b9d2593c8b41f6b82edd156f695149fb0a2370a"
meta-emlinux         = "HEAD:cc2b9983eb2f4e791d8b147ba9aa3579e9e83e13"
meta-emlinux-private = "HEAD:b290dafea47dfbd8df7ee9b04af38c9b73c87b68"

WARNING: /home/build/emlinux/latest-dev/build/../repos/meta-debian/recipes-debian/xz/xz_debian.bb.do_fetch is tainted from a forced run                                                | ETA:  0:00:00
WARNING: /home/build/emlinux/latest-dev/build/../repos/meta-debian/recipes-debian/openssl/openssl_debian.bb.do_fetch is tainted from a forced run
Initialising tasks: 100% |#############################################################################################################################################################| Time: 0:00:00
Sstate summary: Wanted 6 Found 0 Missed 6 Current 433 (0% match, 98% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 2018 tasks of which 2002 didn't need to be rerun and all succeeded.

Summary: There were 2 WARNING messages shown.
```

Auto startup bind server(sysv init)

```
qemuarm64 login: root
root@qemuarm64:~# ps | grep bind
 1607 bind      100m S    /usr/sbin/named -u bind
 2038 root      3220 S    grep bind
```

Auto startup bind server(systemd)

```
root@qemuarm64:~# PAGER=cat systemctl status bind
● bind.service
   Loaded: loaded (/etc/init.d/bind; generated)
   Active: active (running) since Thu 2022-10-06 04:25:01 UTC; 34s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 159 ExecStart=/etc/init.d/bind start (code=exited, status=0/SUCCESS)
    Tasks: 4 (limit: 531)
   Memory: 14.4M
   CGroup: /system.slice/bind.service
           └─176 /usr/sbin/named -u bind

Oct 06 04:24:55 qemuarm64 systemd[1]: Starting bind.service...
Oct 06 04:24:55 qemuarm64 bind[159]: Starting domain name service: namedwrote key file "/etc/bind/rndc.key"
Oct 06 04:25:01 qemuarm64 bind[159]: The specified interface lo is managed by systemd-networkd. Operation refused.
Oct 06 04:25:01 qemuarm64 bind[159]: Please configure DNS settings for systemd-networkd managed interfaces directly in their .network files.
Oct 06 04:25:01 qemuarm64 bind[159]: .
Oct 06 04:25:01 qemuarm64 systemd[1]: Started bind.service.
```

Start and stop bind server(sysv init)

```
root@qemuarm64:~# /etc/init.d/bind stop && /etc/init.d/bind start
Stopping domain name service: named.
Starting domain name service: named.
root@qemuarm64:~# ps | grep bind
 2067 bind      100m S    /usr/sbin/named -u bind
 2074 root      3220 S    grep bind
```

Start and stop bind server(systemd)

```
root@qemuarm64:~# systemctl restart bind
root@qemuarm64:~# PAGER=cat systemctl status bind
● bind.service
   Loaded: loaded (/etc/init.d/bind; generated)
   Active: active (running) since Thu 2022-10-06 04:26:26 UTC; 6s ago
     Docs: man:systemd-sysv-generator(8)
  Process: 220 ExecStart=/etc/init.d/bind start (code=exited, status=0/SUCCESS)
    Tasks: 4 (limit: 531)
   Memory: 15.2M
   CGroup: /system.slice/bind.service
           └─227 /usr/sbin/named -u bind

Oct 06 04:26:26 qemuarm64 systemd[1]: Starting bind.service...
Oct 06 04:26:26 qemuarm64 bind[220]: Starting domain name service: namedThe specified interface lo is managed by systemd-networkd. Operation refused.
Oct 06 04:26:26 qemuarm64 bind[220]: Please configure DNS settings for systemd-networkd managed interfaces directly in their .network files.
Oct 06 04:26:26 qemuarm64 bind[220]: .
Oct 06 04:26:26 qemuarm64 systemd[1]: Started bind.service.
```

dig command result.

```
root@qemuarm64:~# dig www.example.com

; <<>> DiG 9.11.5-P4 <<>> www.example.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 41258
;; flags: qr rd ra; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;www.example.com.               IN      A

;; ANSWER SECTION:
www.example.com.        75950   IN      A       93.184.216.34

;; Query time: 5 msec
;; SERVER: 10.0.2.3#53(10.0.2.3)
;; WHEN: Thu Oct 06 04:01:28 UTC 2022
;; MSG SIZE  rcvd: 60
```

nsloopkup command result.

```
root@qemuarm64:~# ls -la $(which nslookup)
lrwxrwxrwx    1 root     root            22 Oct  6 01:54 /usr/bin/nslookup -> /usr/bin/nslookup.bind
root@qemuarm64:~# nslookup www.example.com
Server:         10.0.2.3
Address:        10.0.2.3#53

Non-authoritative answer:
Name:   www.example.com
Address: 93.184.216.34
Name:   www.example.com
Address: 2606:2800:220:1:248:1893:25c8:1946
```
